MSAL/src/native_auth/public/MSALNativeAuthPublicClientApplication.swift

```diff
@@ -47,11 +47,12 @@ import Foundation
 @objcMembers
 public final class MSALNativeAuthPublicClientApplication: MSALPublicClientApplication {
```

Comment:
@borgesmb I don't like this approach in general. It doesn't feel good. We are adding the `application` class to all the controllers, to the ControllerFactory, to the ResultFactory...

If all that we need is to call `MSALPublicClientApplication` from the `MSALNativeAuthUserAccountResult`, is it possible to instantiate `MSALPublicClientApplication` directly from there?

I see that all we need is the config that is set during the creation of `MSALNativeAuthPublicClientApplication`, but I'd be willing to have it as a global variable accessible by any class. It's not good either, but less intrusive than keeping a copy of the Application class everywhere.

This is what I mean with the global configuration:

```
var globalConfiguration: MSALPublicClientApplicationConfig?

@objcMembers
public final class MSALNativeAuthPublicClientApplication: MSALPublicClientApplication
```

And then instantiate PublicClientApplication directly in the `MSALNativeAuthUserAccountResult`:

In MSALNativeAuthUserAccountResult.swift:
```
    public func getAccessToken(forceRefresh: Bool = false,
                               scopes: [String]? = nil,
                               correlationId: UUID? = nil,
                               delegate: CredentialsDelegate) {

            let msal = try MSALPublicClientApplication(configuration: globalConfiguration!)

            let params = MSALSilentTokenParameters(scopes: scopes ?? [], account: account)
            params.forceRefresh = forceRefresh

            msal.acquireTokenSilent(with: params) { result, error in
               ...
            }
    }
```

Do you think something like this would be possible?

Reply:
Addressed. Please take a look on the new PR; https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/2159